Makefile

```diff
@@ -0,0 +1,210 @@
+REQUIREMENTS_TIMESTAMP = $(VENV)/.requirements.timestamp
+DEV_REQUIREMENTS_TIMESTAMP = $(VENV)/.dev-requirements.timestamp
+
+# general targets timestamps
+TIMESTAMPS = .timestamps
+SYSTEM_PYTHON_TIMESTAMP = $(TIMESTAMPS)/.python-system.timestamp
+PYTHON_LOCAL_BUILD_TIMESTAMP = $(TIMESTAMPS)/.python-build.timestamp
+DOCKER_BUILD_TIMESTAMP = $(TIMESTAMPS)/.dockerbuild.timestamp
+
+# Find all python files that are not inside a hidden directory (directory starting with .)
+PYTHON_FILES := $(shell find ${DJANGO_PROJECT_DIR}/* -type f -name "*.py" -print)
+
+PYTHON_VERSION := 3.7.4
+SYSTEM_PYTHON := $(shell ./getPythonCmd.sh ${PYTHON_VERSION} ${PYTHON_LOCAL_DIR})
+
+# default configuration
+HTTP_PORT ?= 5000
+
+# Commands
+DJANGO_MANAGER := $(DJANGO_PROJECT_DIR)/manage.py
+GUNICORN := $(VENV)/bin/gunicorn
+PYTHON := $(VENV)/bin/python3
+PIP := $(VENV)/bin/pip3
+FLASK := $(VENV)/bin/flask
+YAPF := $(VENV)/bin/yapf
+NOSE := $(VENV)/bin/nose2
+PYLINT := $(VENV)/bin/pylint
+
+
+all: help
+
+
+.PHONY: help
+help:
+	@echo "Usage: make <target>"
+	@echo
+	@echo "Possible targets:"
+	@echo -e " \033[1mSetup TARGETS\033[0m "
+	@echo "- setup              Create the python virtual environment"
+	@echo "- dev                Create the python virtual environment with developper tools"
+	@echo -e " \033[1mFORMATING, LINTING AND TESTING TOOLS TARGETS\033[0m "
+	@echo "- format             Format the python source code"
+	@echo "- lint               Lint the python source code"
+	@echo "- format-lint        Format and lint the python source code"
+	@echo "- test               Run the tests"
+	@echo -e " \033[1mLOCAL SERVER TARGETS\033[0m "
+	@echo "- serve              Run the project using the flask debug server. Port can be set by Env variable HTTP_PORT (default: 5000)"
+	@echo "- gunicornserve      Run the project using the gunicorn WSGI server. Port can be set by Env variable DEBUG_HTTP_PORT (default: 5000)"
+	@echo "- dockerbuild        Build the project localy using the gunicorn WSGI server inside a container"
+	@echo "- dockerrun          Run the project using the gunicorn WSGI server inside a container (exposed port: 5000)"
+	@echo "- shutdown           Stop the aforementioned container"
+	@echo -e " \033[1mCLEANING TARGETS\033[0m "
+	@echo "- clean              Clean genereated files"
+	@echo "- clean_venv         Clean python venv"
+
+
+# Build targets. Calling setup is all that is needed for the local files to be installed as needed.
+
+.PHONY: dev
+dev: $(DEV_REQUIREMENTS_TIMESTAMP)
+
+
+.PHONY: setup
+setup: $(REQUIREMENTS_TIMESTAMP)
+
+
+# linting target, calls upon yapf to make sure your code is easier to read and respects some conventions.
+
+.PHONY: format
+format: $(DEV_REQUIREMENTS_TIMESTAMP)
+	$(YAPF) -p -i --style .style.yapf $(PYTHON_FILES)
+
+
+.PHONY: lint
+lint: $(DEV_REQUIREMENTS_TIMESTAMP)
+	$(PYLINT) $(PYTHON_FILES)
+
+
+.PHONY: format-lint
+format-lint: format lint
+
+
+# Test target
+
+.PHONY: test
+test: $(REQUIREMENTS_TIMESTAMP)
```

Comment:
test should depends on $(DEV_REQUIREMENTS_TIMESTAMP) otherwise nose2 might be missing